code/game/objects/structures/crates_lockers/closets/bsdroppod.dm

```diff
@@ -0,0 +1,86 @@
+//The "BDPtarget" temp visual is created by the expressconsole, which in turn makes two things: a falling droppod animation, and the droppod itself.
+
+
+//------------------------------------BLUESPACE DROP POD-------------------------------------//
+/obj/structure/closet/bsdroppod
+	name = "Bluespace Drop Pod"
+	desc = "It's a storage unit for THE FUTURE NI\[B]\[B]A!!!!!"
+	icon = 'icons/obj/2x2.dmi'
+	icon_state = "BDP"
+	pixel_x = -16//2x2 sprite
+	pixel_y = -5
+	pixel_z = 0
+	layer = TABLE_LAYER//so that the crate inside doesn't appear underneath
+	allow_objects = TRUE
+	allow_dense = TRUE
+	delivery_icon = null
+	can_weld_shut = FALSE
+	armor = list(melee = 30, bullet = 50, laser = 50, energy = 100, bomb = 90, bio = 0, rad = 0, fire = 80, acid = 80)
+	anchored = TRUE
+	anchorable = FALSE
+	var/datum/supply_order/SupplyOrder
+
+/obj/structure/closet/bsdroppod/Initialize(mapload, datum/supply_order/so)
+	. = ..()
+	src.SupplyOrder = so//uses Supply Order passed from expressconsole into BDPtarget
+	addtimer(CALLBACK(src, .proc/open), 30)//open 3seconds after appearing
+
+/obj/structure/closet/bsdroppod/update_icon()//called in initialize, all we have to do is add a door icon
+	cut_overlays()//just incase update_icon somehow gets called multiple times, prevents overlay spam
+	add_overlay("BDP_door")
+	return
+
+/obj/structure/closet/bsdroppod/proc/sparks()
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(5, 1, get_turf(src))
+	s.start()
+
+/obj/structure/closet/bsdroppod/attackby()//only exists for a few seconds, so no need to be welded or anchored or whatever
+	return
+
+/obj/structure/closet/bsdroppod/open(mob/living/user)
+	var/turf/T = get_turf(src)
+	SupplyOrder.generate(T)//not called during populateContents as supplyorder generation requires a turf
+	cut_overlays()
+	add_overlay("BDP_open")
+	playsound(loc, open_sound, 15, 1, -3)
+	sleep(30)
+	sparks()//wait 3 seconds, then make some sparks and delete
+	qdel(src)
+
+//------------------------------------FALLING BLUESPACE DROP POD-------------------------------------//
+/obj/effect/temp_visual/BDPfall
+	icon = 'icons/obj/2x2.dmi'
+	icon_state = "BDP_falling"
+	pixel_x = -16
+	pixel_y = -5
+	pixel_z = 200
```

Comment:
Er... sure this does anything?

Reply:
"This displaces the object's icon vertically by the specified number of pixels. This is meant to be used in situations where world.map_format is used to display something other than a top-down form, for instance in an isometric or side-view display. In a top-down mode pixel_z behaves the same as pixel_y, except that it does not rotate with changes to client.dir.

This effect is purely visual and does not influence such things as movement bumping or view() range calculations."

Reply:
yes